Exiled.API/Features/Map.cs

```diff
@@ -229,6 +253,189 @@ public static Room FindParentRoom(GameObject objectInRoom)
+        /// <param name="allowRecall">Sets this ragdoll as respawnable by SCP-049.</param>
+        /// <returns>The Ragdoll component (requires Assembly-CSharp to be referenced).</returns>
+        public static global::Ragdoll SpawnRagdoll(Player victim, DamageTypes.DamageType deathCause, Vector3 position, Quaternion rotation = default, Vector3 velocity = default, bool allowRecall = true)
+        {
+            return SpawnRagdoll(
+                        victim.Role,
+                        deathCause,
+                        victim.DisplayNickname,
+                        position,
+                        rotation,
+                        velocity,
+                        allowRecall,
+                        victim.Id,
+                        victim.GameObject.GetComponent<Dissonance.Integrations.MirrorIgnorance.MirrorIgnorancePlayer>().PlayerId);
+        }
+
+        /// <summary>
+        /// Spawns a ragdoll on the map based on the different arguments.
+        /// </summary>
+        /// <remarks>
+        /// Tip: You can do '<paramref name="allowRecall"/>: true, <paramref name="playerId"/>: MyPlayer.Id' to skip parameters.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// // Code to spawn a fake ragdoll
+        /// if (ev.Player == MyPlugin.TheInmortalPlayer)
+        /// {
+        ///     var fakeRagdoll = Map.SpawnRagdoll(ev.Player.Role, ev.Player.Position, deathCause:DamageTypes.Fall, victimNick: ev.Player.DisplayNickname, playerId: ev.Player.Id);
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="roleType">The <see cref="RoleType"/> to use as ragdoll.</param>
+        /// <param name="deathCause">The death cause, expressed as a <see cref="DamageTypes.DamageType"/>.</param>
+        /// <param name="victimNick">The name from the victim, who the corpse belongs to.</param>
+        /// <param name="position">Where the ragdoll will be spawned.</param>
+        /// <param name="rotation">The rotation for the ragdoll.</param>
+        /// <param name="velocity">The initial velocity the ragdoll will have, as if it was exploded.</param>
+        /// <param name="allowRecall">Sets this ragdoll as respawnable by SCP-049. Must have a valid <paramref name="playerId"/>.</param>
+        /// <param name="playerId">Used for recall. The <see cref="Player.Id"/> to be recalled.</param>
+        /// <param name="mirrorOwnerId">The <see cref="Dissonance.Integrations.MirrorIgnorance.MirrorIgnorancePlayer"/>'s PlayerId field, likely used in the client.</param>
+        /// <param name="damage">Usually ignored, the killing blow damage.</param>
+        /// <param name="killerName">Usually ignored, the killer's name.</param>
+        /// <param name="killerId">Usually ignored, the killer's ID.</param>
+        /// <returns>The Ragdoll component (requires Assembly-CSharp to be referenced).</returns>
+        public static global::Ragdoll SpawnRagdoll(
```

Comment:
What's point of this method? It has 11 arguments and 4 of which are just unpacked arguments from `PlayerStats.HitInfo`. If the game devs will decide to add more than 4 arguments for the `PlayerStats.HitInfo`'s constructor, it'll break this method.

Reply:
Right, didn't consider that detail, maybe I should have used the `new PlayerStats.HitInfo() {}` method so it's forwards-compatible *(if that word even exist)*.

Either way, the point of this method is to avoid having to set the damage and other unused stuff, so you don't have to create the entire object with most of its members ignored anyways. I could have perfectly left the "deathCause" value by itself and it would have had the same purpose.

Reply:
You already made an overload which requires a Player instance, if someone wants spawn a "fully controlled" ragdoll, so they forced to use `SpawnRagdoll(RoleType, string, Vector3, Quaternion, Vector3, PlayerStats.HitInfo, bool, int, string)`, another way looks so useless as a middle level.